### ISSUE TITLE
Orbits in 3D with inclinations?

### ISSUE THREAD

**Issue by expenses** [atbentley/bevy_orbits#1]
Hi! I've been using a modified version of [math.rs](https://github.com/atbentley/bevy_mod_orbits/blob/main/src/math.rs) for orbits in my project and I've found it to be very useful. Ideally I want to be able to calculate orbits for bodies in 3D space with a set inclination. I haven't looked into keplers laws too much though, do you know how easy this would be to add?

**Comment by atbentley** [comment]
Hey @expenses! I haven't dived into the world of 3D orbits yet, but I don't think it should be too difficult. Although I think it would take a bit of trial and error because I can find a good source.

As I understand it we would need to introduce two more parameters, the inclination and the longitude of the ascending node.

This image describes it the best that I've seen: https://commons.wikimedia.org/wiki/File:Orbital_elements.svg

The inclination would then be the rotation that shifts the orbit out of the orbital plane. It rotates around the line that lies on the orbital plane, passes through the parent body, and is perpendicular to the long axis of the orbital ellipse. The rotation is applied such that the furthest point of the orbit moves upwards. So that seems ok.

The longitude of the ascending node then looks to be another rotation, it looks like it is applied after the inclination and looks to be a rotation about the axis that is perpendicular to the parent body's orbital plane (so 'up').

What is confusing to me is what order all of these rotations should be applied in. There are only a small number of options though, so we could just try them all :D

Were you interested in giving that a shot or do you want to wait until I get around to it? It is something that I've been wanting to do, I'm not super focused on this project right now but I will come back to it at some point.

**Comment by expenses** [comment]
Right, cool, I think I have enough information to look into this soon! In my own words, I think we're just doing one rotation around an **axis that lies on the orbital plane**. So for example an orbit is inclined so that the -x side goes up towards +y and the +x side goes down towards -y, we're rotating on z axis. I'd [just reach for a quaternion for this](https://docs.rs/glam/latest/glam/f32/struct.Quat.html#method.from_axis_angle) but they're probably not strictly needed. I'll do some testing tomorrow by appending rotations onto the output of the `calculate_position` function.

**Comment by atbentley** [comment]
Let me know how it goes!
